src/nbl/CMakeLists.txt

```diff
@@ -319,6 +319,11 @@ set(NBL_VIDEO_SOURCES
 # CUDA
 	${NBL_ROOT_PATH}/src/nbl/video/CCUDAHandler.cpp
 	${NBL_ROOT_PATH}/src/nbl/video/CCUDADevice.cpp
+
+# JSON
+	${NBL_ROOT_PATH}/src/nbl/video/device_limits.json
+	${NBL_ROOT_PATH}/src/nbl/video/device_features.json
```

Comment:
@AnastaZIuk is this even a right thing to do?

Reply:
of course it's invalid, currently they seem to be inputs for combined [gen.py](https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#diff-f616d789839c4626b1374c0c82bf01e4bcad873788ee702cbe30f1ff50bbaab0) and [DeviceGen.py](https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#diff-59690db9fdfbd1f813e5be71e8a42c06c649d8a903e0ed5a82b0831716a36ecd) python scripts, they should be handled with [custom target](https://cmake.org/cmake/help/latest/command/add_custom_target.html) + [custom command](https://cmake.org/cmake/help/latest/command/add_custom_command.html) pair, those json files should be flagged as `DEPENDS` of the custom command. I guess outputs are knows as well - they should be passed to the custom command's `OUTPUTS` and custom target's `DEPENDS`

Reply:
also handled with python or not - those scripts should never hardcode location to the inputs like do do currently, the script should take an input and expected output + be executed by a custom command which prepares the execution of the script given correct arguments for the script execution

Reply:
Done

Reply:
@AnastaZIuk mask as resolved if done